openpype/hosts/houdini/plugins/publish/validate_primitive_hierarchy_paths.py

```diff
@@ -104,3 +113,60 @@ def get_invalid(cls, instance):
                 "(%s of %s prims)" % (path_attr, len(invalid_prims), num_prims)
             )
             return [output_node.path()]
+
+    @classmethod
+    def repair(cls, instance):
+        """Add a default path attribute Action.
+
+        it's a helper action more than a repair action,
+        which used to add a default single value.
+        """
+
+        output_node = instance.data.get("output_node")
+        rop_node = hou.node(instance.data["instance_node"])
+
+        path_attr = rop_node.parm("path_attrib").eval()
+
+        frame = instance.data.get("frameStart", 0)
+        geo = output_node.geometryAtFrame(frame)
+
+        paths = geo.findPrimAttrib(path_attr) and \
+                    geo.primStringAttribValues(path_attr)
```

Comment:
It'd likely be easier to maintain to just do:
```python
if not cls.get_invalid(instance):
    return
```

Reply:
It's a nice idea. 
I think `get_invalid` will need to be refactored as the current logic doesn't support that. 

I think `add path action` button should only work if these checks evaluate to true 
`not attrib`
`not attrib.strings()`
`invalid_prims` 

and they do return `output_node.path()`  other checks return `None` or `rop_node.path()` 
so, this condition may be the one 
as well as moving logs to another method something similar to `get_invalid`  and `get_invalid_with_message`

```python
if  cls.get_invalid(instance) != [output_node.path()] :
    return
```

Reply:
I think your check makes sense (with an added comment as to why we're checking that specifically.)
However, still be aware that `output_node` could be not set and thus `.path()` might not be available. ;) Which gets us into the unfortunate cases where we still need to perform additional checks.

Reply:
well, I removed `.path()` which is more consistent with `SelectInvalidAction` 
also, if `output_node` could be not set, it will just be `None`